pulp_rpm/app/tasks/publishing.py

```diff
@@ -15,13 +16,74 @@
 
 from pulpcore.plugin.tasking import WorkingDirectory
 
-from pulp_rpm.app.models import Package, RpmPublication, UpdateRecord
+from pulp_rpm.app.models import Package, RepoMetadataFile, RpmPublication, UpdateRecord
 
 log = logging.getLogger(__name__)
 
 REPODATA_PATH = 'repodata'
 
 
+class PublicationData:
+    """
+    Encapsulates data relative to publication.
+
+    """
+
+    def __init__(self, publication):
+        """
+        Setting Publication data.
+
+        """
+        self.publication = publication
+        self.packages = []
+        self.repomdrecords = []
+
+        self.populate()
+        self.download_metadata_files()
+
+    def download_metadata_files(self):
```

Comment:
Maybe prepare_metadata_files?

Reply:
sounds good to me

Reply:
I like the comment from your suggestion: Copies metadata files from the Artifact **storage**.